### PR TITLE
Added support for decimal and partially rewrote insert and update form generation code.

### LINKED ISSUE #54: Abstract out the getInputBlock method from insert to be used by filter, insert and update.
Needs to be done to avoid copying and pasting the code in 3 different locations and having to manage them.

### LINKED ISSUE #72: Add support for `double` data type

### LINKED ISSUE #74: Format bug with seconds represented in `time` type
image

### LINKED ISSUE #81: Picker not incrementing properly for `float` type on insert/update
Currently utilizes same as `int`

### PR DESCRIPTION
**Pull request by Synicix**
- Update Readme to be a bit more clear on dealing with git submodule 
- Split the label and input block generation code and moved it to TableAttributes and its child #
- Started to unify some of the CSS shared between Insert and Update (Somehwhat fix #54 )
- Fix #81 
- Fix #74 
- Fix #72

### CODE REVIEW

**Comment by Synicix** [comment]
Planning to tackle a few more issue along with this pull request due to it being close to the input form generation code. Will mark this as ready for review once other issues resolved

**Comment by Synicix** [comment]
Need to update change log, please wait for that commit

**Comment by Synicix** [comment]
Change logs added

**Review by guzman-raphael — CHANGES_REQUESTED** [review]
@mahos Please make sure to verify to prevent conflicts. Specifically since @Synicix has created a new subdirectory structure for `InsertTuple`, `DeleteTuple`, etc.

@maho @Synicix For PR's that make visual, user-facing updates, we should adopt a policy of including minimally some screenshots/GIF's that demonstrate the fix. This provides meaningful docs, discussion points, and relevant proof-of-work while we don't currently have integration tests for frontend side of the stack. Here is a [good reference](https://medium.com/statuscode/make-your-pull-requests-visual-for-front-end-changes-ef91d83d4fc2) I came across recently to reduce any overhead with generating these screenshots/GIF's.

@Synicix Please add some screenshot/GIF's pertaining to:
- Picker increment on `float` for `insert`, `update`
- Correctly displayed seconds in `time` type
- Support for `double` type. FYI, I just refreshed `alpha_company` in our test DB to now include a `double` type in table `Computer`, attribute `computer_processor` .

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and
 ## [Unreleased]
 ### Changed
 - Modify dark and default logos to reflect new `DataJoint LabBook` name. PR #70
+- Refactor Insert and Update component code to share common form generation code
+- Added support for double data type
+- Fixed issue with input fields for decimal, floats, and double not having the correct step settings.
+- Updated formating for time to match HH:MM:SS, before it would chop off MM and SS if was less then 10
```
[the remark this answers is not in the document]
Comment (Synicix) [reply CHANGELOG.md:11]: Cool will update Change logs to match that convention

**Comment by Synicix** [comment]
Time how display the leading 0:
image

**Comment by Synicix** [comment]
double support added, no longer blows up. Also the behavior for float, and double now has a step='any' which basically allow the stepper to go by 1 up and down but can take any decimal number

Inserting double:
image

After clicking step up:
image

**Review by guzman-raphael — DISMISSED** [review]

File: src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx [hunk 2]
```diff
@@ -68,218 +70,246 @@ class TableAttribute {
...
-          <div className="fieldUnit" key={JSON.stringify(tableAttribute)}>
-            <div className="dateTimeFields">
-              <input type='date' defaultValue={defaultValue} id={tableAttribute.attributeName + '__date'} onChange={handleChange.bind(this, tableAttribute.attributeName + '__date')}></input>
-              <input type='time' step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + '__time'} onChange={handleChange.bind(this, tableAttribute.attributeName + "__time")}></input>
-            </div>
+      else if (tableAttribute.attributeType === TableAttributeType.DATETIME || tableAttribute.attributeType === TableAttributeType.TIMESTAMP) {
+        return(
+          <div className="dateTimeFields">
+            <input type="date" defaultValue={defaultValue} id={tableAttribute.attributeName + "__date"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__date")}></input>
+            <input type="time" step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + "__time"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "time")}></input>
```
Comment (mahos) [inline src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:289]: onChange for the time input field probably should be
```suggestion
            <input type="time" step="1" defaultValue={defaultValue} id={tableAttribute.attributeName + "__time"} onChange={(e) => handleChange(e, tableAttribute.attributeName + "__time")}></input>
```

Comment (Synicix) [reply src/Components/MainTableView/DataStorageClasses/TableAttribute.tsx:289]: opps screwed on the renaming. thanks nice catch

**Review by guzman-raphael — APPROVED** [review]

**Review by mahos — APPROVED** [review]

**Comment by maho** [comment]
By the way, I was mentioned by accident I guess, instead of @mahos

**Pull request merged**